src/main/js/store/actions/globalConfiguration.js

```diff
@@ -258,17 +304,29 @@ export function updateConfig(config) {
             request = ConfigRequestBuilder.createNewConfigurationRequest(ConfigRequestBuilder.CONFIG_API_URL, csrfToken, config);
         }
         request.then((response) => {
-            response.json()
-                .then((responseData) => {
-                    if (response.ok) {
-                        const newId = responseData.id;
-                        const updatedConfig = FieldModelUtilities.updateFieldModelSingleValue(config, 'id', newId);
-                        dispatch(configUpdated(updatedConfig));
-                        dispatch(refreshConfig(newId));
-                    } else {
-                        handleFailureResponse(dispatch, responseData, response.status);
-                    }
-                });
+            if (response.ok) {
+                if (response.status === 201) {
+                    response.json()
+                        .then((responseData) => {
+                            const newId = responseData.id;
+                            const updatedConfig = FieldModelUtilities.updateFieldModelSingleValue(config, 'id', newId);
+                            dispatch(configUpdated(updatedConfig));
+                            dispatch(refreshConfig(newId));
+                        });
+                } else {
+                    dispatch(refreshConfig(id));
+                }
```

Comment:
You could remove the else (for the response.status check) and only need to call refresh once: 
```suggestion
                        });
                }
                dispatch(refreshConfig(id));
```

Reply:
Ah, but you'll have to create an id variable outside of the if-statement and set it for a 201.